docs/Changelog.md

```diff
@@ -15,6 +15,7 @@
   * [#3709](https://github.com/TouK/nussknacker/pull/3709) `BestEffortJsonSchemaEncoder` encodes only Strings for `"type": String`
   * [#3730](https://github.com/TouK/nussknacker/pull/3730) Additional fields are not trimmed during encoding when `additionalProperties` are allowed by schema
   * [#3742](https://github.com/TouK/nussknacker/pull/3742) Improvements: Json Schema more strict encoding - matching to schema validate
+  * [#3749](https://github.com/TouK/nussknacker/pull/3749) Improvements: JsonSchema better encode integer schema
```

Comment:
It's not 'better', it just has better error messages?

Reply:
It's also better, because of:
`
    def encodeBigDecimalToIntegerSchema(bigDecimalValue: BigDecimal): Validated[NonEmptyList[String], Json] = Try(bigDecimalValue.toLongExact) match {
      case Failure(_) => error(value.getClass.getName, schema.toString, fieldName)
      case Success(v) => Valid(jsonEncoder.encode(v))
    }
`
If number is proper long value then we encode value to long, e.g. 1.0 will be encoded to 1L.